packages/expo-polyfills/src/index.ts

```diff
@@ -176,42 +176,46 @@ class SyncSecureStorage implements Storage {
 
   constructor() {
     this.storage = new Map<string, string>();
-    this.initPromise = new Promise((resolve, reject) => {
-      // Fetch ::keys::
-      SecureStore.getItemAsync('___keys___')
-        .then((keysStr) => {
-          if (!keysStr) {
-            // No keys currently, just resolve
-            this.initialized = true;
-            resolve();
-            return;
-          }
-          // Parse keys
-          const keys = JSON.parse(keysStr) as string[];
-          const promises = keys.map((key) => {
-            return new Promise<[string, string | null]>((resolve, reject) => {
-              SecureStore.getItemAsync(key)
-                .then((val) => {
-                  resolve([key, val]);
-                })
-                .catch(reject);
-            });
-          });
-          Promise.all(promises)
-            .then((values) => {
-              for (const [key, value] of values) {
-                if (!value) {
-                  continue;
-                }
-                this.storage.set(key, value);
-              }
-              this.initialized = true;
-              resolve();
-            })
-            .catch(reject);
-        })
-        .catch(reject);
-    });
+    this.initPromise = this.initStorage();
+  }
+
+  private async initStorage(): Promise<void> {
```

Comment:
The init logic before was admittedly very confusing with the nested promises

Thank god for async fns